core/shared/src/main/scala/fs2/Chunk.scala

```diff
@@ -694,22 +694,26 @@ object Chunk extends CollectorK[Chunk] with ChunkCompanionPlatform {
 
   /** Creates a chunk backed by an array. */
   def array[O](values: Array[O]): Chunk[O] =
-    values.size match {
+    array(values, 0, values.length)
+
+  /** Creates a chunk backed by an array. */
+  def array[O](values: Array[O], offset: Int, length: Int): Chunk[O] =
+    length match {
       case 0 => empty
-      case 1 => singleton(values(0))
+      case 1 => singleton(values(offset))
       case _ =>
         values match {
-          case a: Array[Boolean] => booleans(a)
-          case a: Array[Byte]    => bytes(a)
-          case a: Array[Short]   => shorts(a)
-          case a: Array[Int]     => ints(a)
-          case a: Array[Long]    => longs(a)
-          case a: Array[Float]   => floats(a)
-          case a: Array[Double]  => doubles(a)
-          case _                 => boxed(values)
+          case a: Array[Boolean] => booleans(a, offset, length)
+          case a: Array[Byte]    => bytes(a, offset, length)
+          case a: Array[Short]   => shorts(a, offset, length)
+          case a: Array[Int]     => ints(a, offset, length)
+          case a: Array[Long]    => longs(a, offset, length)
+          case a: Array[Float]   => floats(a, offset, length)
+          case a: Array[Double]  => doubles(a, offset, length)
+          case a: Array[Char]    => chars(a, offset, length)
```

Comment:
The case for `Char` was missing here, resulting in `boxed` getting called for `Array[Char]`, which would then cause failures on Scala.js.